synapse/federation/transaction_queue.py

```diff
@@ -463,7 +463,19 @@ def _transaction_transmission_loop(self, destination):
                 # pending_transactions flag.
 
                 pending_pdus = self.pending_pdus_by_dest.pop(destination, [])
+
+                # We can only include at most 50 PDUs per transactions
+                pending_pdus, leftover_pdus = pending_pdus[:50], pending_pdus[50:]
+                if leftover_pdus:
+                    self.pending_pdus_by_dest[destination] = leftover_pdus
```

Comment:
This makes it non-atomic -- it should be `self.pending_pdus_by_dest[destination].extend(leftover_pdus)`, maybe?

Reply:
I think by "atomic" it means "doesn't yield", tbh

Reply:
if potential threading isn't a concern, then, looks fine to me.

Reply:
> I think by "atomic" it means "doesn't yield", tbh

this, ftr